l10n_ch_payment_slip/payment_slip.py

```diff
@@ -480,10 +481,30 @@ def _draw_address(self, canvas, print_settings, initial_position, font,
         text.setFont(font.name, font.size)
         text.textOut(com_partner.name)
         text.moveCursor(0.0, font.size)
-        for line in com_partner.contact_address.split("\n"):
-            if not line:
-                continue
-            text.textLine(line)
+        pt_cpny_add = print_settings.bvr_print_cpny_address
+        
+        if pt_cpny_add == 'full':
+            for line in com_partner.contact_address.split("\n"):
+                if not line:
+                    continue
+                
+                text.textLine(line)
+        else:
+            city_line = (com_partner.zip or '') + ' ' + (com_partner.city or '') + ' ' + (com_partner.state_id.code or '')
+            if pt_cpny_add == 'first_line_only':
+                if com_partner.street:
+                    line = com_partner.street
+                    text.textLine(line)
+                    
+                text.textLine(city_line)
+                
+            if pt_cpny_add == 'second_line_only':
+                if com_partner.street2:
+                    line = com_partner.street2
+                    text.textLine(line)
+         
+                text.textLine(city_line)
```

Comment:
No we have usually Address in street and CP in street2 if there is any.
The original modifications offered to print "full address", "only line 1" (street) or "no address" but I added "only line 2" (street 2) just in case.

I guess it would be better to add a new bvr address but that would make information redondant in most cases. That is why I sorted it out like that.

I understand your point and yes I will add the street selection in a custom module.